### DISCUSSION TITLE
What is the correct way to polyfill structuredClone?

### DISCUSSION THREAD

**Discussion by dragonlobster** [babel/babel#15619] [Q&A] [answered]
Hello, I am attempting to transpile code to ES5 and want to use structuredClone, however I'm in an runtime environment does not support it (goja). So I wish to polyfill this function, could someone point me in the right direction on how to achieve this. Thank you

**Comment by JLHwung** [comment, marked as answer]
You can try https://github.com/ungap/structured-clone.

**Reply by pedroSoaresll** [reply]
We can find an indication by the MDN documentation at this link: https://developer.mozilla.org/en-US/docs/Web/API/structuredClone#see_also

![image](https://github.com/babel/babel/assets/11558773/08fb1ff8-8b8b-48e3-9e0f-1b9d03b2bf3b)

As @JLHwung commented, you can try https://github.com/ungap/structured-clone but you also can try https://github.com/zloirock/core-js

IMHO https://github.com/zloirock/core-js looks like a better option because of the community behind it.

**Reply by paulmelero** [reply]
Unfortunately they don't offer support for `File` yet!

**Comment by zloirock** [comment]
It's available from `core-js@3.20`, so it should be available, for example, in Babel with `preset-env` with proper settings.
